File: app/src/main/java/xyz/sleekstats/completist/model/ShowCredits.java
package xyz.sleekstats.completist.model;

import com.google.gson.annotations.SerializedName;

import java.util.HashSet;
import java.util.Set;

public class ShowCredits {

    @SerializedName("cast")
    private Set<ShowByPerson> cast;

    @SerializedName("crew")
    private Set<ShowByPerson> crew;

    private Set<ShowByPerson> both;

    public Set<ShowByPerson> getCrew() {
        return filterCrew(crew);
    }

    public Set<ShowByPerson> getCast() {
        return cast;
    }

    public Set<ShowByPerson> bothLists() {
        if(both == null) {
            Set<ShowByPerson> ShowByPersonSet = new HashSet<>(cast);
            ShowByPersonSet.addAll(getCrew());
            both = ShowByPersonSet;
        }
        return both;
    }

    private Set<ShowByPerson> filterCrew(Set<ShowByPerson> unfiltered) {
        Set<ShowByPerson> filteredSet = new HashSet<>();
        for (ShowByPerson film : unfiltered) {
            if (film.getJob().equals("Director") || film.getJob().equals("Writer") || film.getJob().equals("Screenplay")) {
                filteredSet.add(film);
            }
        }
        return filteredSet;
    }
}

File: app/src/main/java/xyz/sleekstats/completist/model/ShowCredits.kt
package xyz.sleekstats.completist.model

import com.google.gson.annotations.SerializedName
import java.util.*
import kotlin.collections.HashSet

class ShowCredits {
    @SerializedName("cast")
    val cast: Set<ShowByPerson> = emptySet()

    @SerializedName("crew")
    private val crew: Set<ShowByPerson> = emptySet()
        get() = filterCrew(field)

    private var both: Set<ShowByPerson>? = null

    fun bothLists(): Set<ShowByPerson>? {
        if (both == null) {
            val showByPersonSet: MutableSet<ShowByPerson> = HashSet(cast)
            showByPersonSet.addAll(crew)
            both = showByPersonSet
        }
        return both
    }

    private fun filterCrew(unfiltered: Set<ShowByPerson>): Set<ShowByPerson> = unfiltered.filterTo(HashSet()) { filteredCrewJobs.contains(it.job) }

    companion object {
        private val filteredCrewJobs = arrayOf("Director", "Writer", "Screenplay")
    }
}